client/src/client.rs

```diff
@@ -986,13 +986,22 @@ impl RpcApi for Client {
     ) -> Result<T> {
         let req = self.client.build_request(&cmd, &args);
         if log_enabled!(Debug) {
-            debug!("JSON-RPC request: {}", serde_json::to_string(&req).unwrap());
+            debug!(target: "bitcoincore_rpc", "JSON-RPC request: {} {}", cmd, serde_json::Value::from(args));
```

Comment:
Why not just use `args` instead of `serde_json::Value::from(args)`?

Reply:
I wanted this to be printed as a compact json string and not as a `Debug` string. `args` would have to be formatted with `{:?}`, which would also use `Debug` on the nested `Value`s instead of stringifying them to json.

Reply:
It might be be more performant to stringify each argument individually and join them with `, ` instead of creating a new `Value` for this, but I thought it was acceptable because its behind a `log_enabled!` check.

Reply:
Hmm yeah not sure how that `from` works. It looks like it would create a vector cloning all items inside. Which would be quite bad. But I'd have to check that.

How about `args.iter().fold(false, |c,e| if c { write!(f, ", ").unwrap(); } write!(f, "{}", e).unwrap(); } true);`?